fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordCursorContinuation.java

```diff
@@ -60,6 +62,16 @@
     @Nullable
     byte[] toBytes();
 
+    /**
+     * Serialize this continuation to a ByteString object.
+     * @return a (possibly EMPTY) ByteString containing a binary serialization of this continuation
+     */
+    @Nonnull
+    default ByteString toByteString() {
+        final byte[] bytes = toBytes();
+        return bytes == null ? ByteString.EMPTY : ByteString.copyFrom(bytes);
```

Comment:
It may make the most sense to have @alecgrieser to the actual review, but one comment upon re-reading the original issue is:

> and it would also be essentially free for many of the cursors (in particular, the ones built on top of protobuf messages)

So, I think probably the intent was that this can be a good default implementation, but to look through the subclass of cursors and identify those that already have their continuation encoded as a `ByteString` and just natively return it.

Reply:
I added the overrides for all the implementations that have ```cachedBytes``` encoded from the proto message, by adding a ```cachedByteString``` variable for them, and using the same way to lazily populate the cached ByteString object and return it. Please let me know if it looks good to you. @scgray

Reply:
I think that this approach, caching the the byte string if possible but having the default make a copy, is the only way that doesn't change the interface, so this approach is probably fine.